tf2/src/cache.cpp

```diff
@@ -261,7 +261,10 @@ bool TimeCache::insertData(const TransformStorage& new_data)
       break;
     storage_it++;
   }
-  storage_.insert(storage_it, new_data);
+  if (storage_it != storage_.end() && storage_it->stamp_ == new_data.stamp_)
+    *storage_it = new_data;
```

Comment:
As discussed in #414 I'd rather see the newer data dropped instead of overwriting the older data. With the current semantics `/tf` data is effectively considered immutable. And without the ability to rewrite sections of data consistently you could easily end up with data from two datasources being improperly interpolated between.

Reply:
OK.  I changed the patch to return immediately when a duplicate timestamp is seen instead of overwriting data.

Reply:
In simply changing the assignment to use the old value, one of the tests in test_tf2 was failing. Further analysis showed that it was not anchoring the forward and inverse transforms to the first frame in the list but the second one.  

So, the new code for rejecting redundant transforms actually caught a bug in the tests. :cake:

Reply:
Great thanks!